### PR TITLE
[CF1] per app vpn

### PR DESCRIPTION
**Pull request by deadlypants1973**
Merge on MONDAY DECEMBER 23, 2024 ONLY

### Summary

14161

### Screenshots (optional)

tbd

### Documentation checklist

- [ ] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.

### CODE REVIEW

**Comment by j-bibodi** [comment]
I have 3 comments:

1. From my observation in the past, we have not attached screenshots for configuring the steps. I am much in favor of adding the screenshots but based on @kkrum approval. 

2. Should we also add how to get the package name or app identifier for a specific app? This will help admins to not make any mistakes while entering the tunneled app package name. For example: we get the package name or app identifier from the ID query param in the Play store URL - https://play.google.com/store/apps/details?id=com.cloudflare.cloudflareoneagent

3. The JSON object shared in the example shows Chrome browser and Gmail app package name, would it be ok to publish the 3rd party app identifier in our documentation? I just want to explicitly callout and get approval for the same.
```
{
       "key": "app_identifier",
       "valueString": "com.android.chrome" # Application package name/unique bundle identifier for the Chrome app browser
},
```

Otherwise, LGTM!

**Comment by deadlypants1973** [comment]
> I have 3 comments:
> 
> 1. From my observation in the past, we have not attached screenshots for configuring the steps. I am much in favor of adding the screenshots but based on @kkrum approval.
> 2. Should we also add how to get the package name or app identifier for a specific app? This will help admins to not make any mistakes while entering the tunneled app package name. For example: we get the package name or app identifier from the ID query param in the Play store URL - https://play.google.com/store/apps/details?id=com.cloudflare.cloudflareoneagent
> 3. The JSON object shared in the example shows Chrome browser and Gmail app package name, would it be ok to publish the 3rd party app identifier in our documentation? I just want to explicitly callout and get approval for the same.
> 
> ```
> {
>        "key": "app_identifier",
>        "valueString": "com.android.chrome" # Application package name/unique bundle identifier for the Chrome app browser
> },
> ```
> 
> Otherwise, LGTM!

1. No screenshots is favored by docs team as well because dashboard screenshots like this often go out of date very fast and it's better to have only text instructions rather than a confusingly outdated screenshot. I will remove them all.

2. I think that would be helpful. Could you write something and tell me where to include it in docs?

3. It's meant to be an example of a possible value (in this case, Chrome), right? I don't see an issue with it.

File: src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx [hunk 1]
```diff
@@ -44,6 +44,13 @@ Refer to the [generic instructions for iOS](/cloudflare-one/connections/connect-
 
 Intune allows you to insert [predefined variables](https://learn.microsoft.com/en-us/mem/intune/apps/app-configuration-policies-use-ios#tokens-used-in-the-property-list) into the XML configuration file. For example, you can set the [`unique_client_id`](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/parameters/#unique_client_id) key to `{{deviceid}}` for a [device UUID posture check](/cloudflare-one/identity/devices/warp-client-checks/device-uuid/) deployment.
 
+### Per-app VPN for iOS
+
+Before proceeding with per-app VPN configuration, you must make sure Autoconnect is disabled for your organization in the Cloudflare dashboard.
```
Comment (deadlypants1973) [inline src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx]: question

File: src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx [hunk 2]
```diff
@@ -44,6 +44,47 @@ Refer to the [generic instructions for iOS](/cloudflare-one/connections/connect-
 
 Intune allows you to insert [predefined variables](https://learn.microsoft.com/en-us/mem/intune/apps/app-configuration-policies-use-ios#tokens-used-in-the-property-list) into the XML configuration file. For example, you can set the [`unique_client_id`](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/parameters/#unique_client_id) key to `{{deviceid}}` for a [device UUID posture check](/cloudflare-one/identity/devices/warp-client-checks/device-uuid/) deployment.
 
+### Per-app VPN for iOS
```
Comment (j-bibodi) [inline src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx:47]: I think we should document that per-app VPN is supported on Cloudflare One Agent app version 1.8 or greater for iOS.

File: src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx [hunk 3]
```diff
@@ -122,3 +163,128 @@ To deploy WARP on Android devices:
     3. Select **Review + save** > **Save**.
 
 Intune will now deploy the Cloudflare One Agent to user devices.
+
+### Per-app VPN for Android
```
Comment (j-bibodi) [inline src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx:182]: I think we should document that per-app VPN is supported on Cloudflare One Agent app version 2.3 or greater for Android.

File: src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx [hunk 4]
```diff
@@ -122,3 +174,134 @@ To deploy WARP on Android devices:
...
+	2. Select the Platform as **Android Enterprise**.
+	3. Select the desired **Profile Type**. (For example: Personally-Owned Work Profile Only)
+	4. Select **Cloudflare One Agent** as the **Targeted app**.
+	5. Select **Next**.
+
+3. Fill out the settings for the configuration policy.
+	1. Select **Configuration setting format** as **Enter JSON data**.
+	2. Enter your desired deployment parameters in the `managedProperty` field. For example:
+
+	```sh
```
Comment (j-bibodi) [inline src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx]: Should we align this example to the first word of the point 3.2 i.e, `Enter` ?

Right now, I see this example is aligned with point number 3. Maybe, I am just being picky here! Feel free to ignore if we follow this standard format.

File: src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/jamf.mdx [hunk 5]
```diff
@@ -66,3 +66,48 @@ Create an [XML file](/cloudflare-one/connections/connect-devices/warp/deployment
 8. Select **Save**.
 
 Jamf is now configured to deploy the Cloudflare One Agent.
+
+### Per-app VPN
```
Comment (j-bibodi) [inline src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/jamf.mdx:70]: We should have this note in JAMF as well.

:::note
Per-app VPN is supported on Cloudflare One Agent version `1.8` or greater for iOS.
:::

File: src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx [hunk 6]
```diff
@@ -122,3 +178,134 @@ To deploy WARP on Android devices:
...
+									"key": "warp_tunnel_protocol",
+									"valueString": "wireguard"
+								}
+							]
+						}
+					]
+				}
+			]
+		}
+		```
```
Comment (deadlypants1973) [inline src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx:299]: @ ranbel A new sentence here on a new PR stating:

"
Refer to [Per-app VPN parameters](/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/parameters/#per-app-vpn-parameters-android) to learn more about the MDM parameters introduced to support the per-app VPN for Android devices.
"

after the [parameters PR](https://github.com/cloudflare/cloudflare-docs/pull/18847) and this PR have been merged.

**Review by Maddy-Cloudflare — APPROVED** [review]

File: src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx (hunk 6 above)
Comment (ranbel) [reply src/content/docs/cloudflare-one/connections/connect-devices/warp/deployment/mdm-deployment/partners/intune.mdx:299]: added to https://github.com/cloudflare/cloudflare-docs/pull/18896

**Pull request merged**